crates/core/src/chunk.rs

```diff
@@ -67,7 +63,7 @@ impl ChunkInfo {
         hasher.update(self.prev_state_root.as_ref());
         hasher.update(self.post_state_root.as_slice());
         hasher.update(withdraw_root.as_slice());
-        hasher.update(self.data_hash.as_slice());
+        hasher.update(self.msg_queue_hash.as_slice()); // FIXME: is this correct?
```

Comment:
does pi hash struct correct? @roynalnaruto

Reply:
Usage of `data_hash` is correct until euclid phase1. From phase2 onwards, we have those changes (described in the notion doc).

So my suggestion is not to update this branch `chore/revm-v55-upgrade` directly since we depend on this branch as a patch for euclid phase1 also.

Instead we can publish a tag for sbv (from `chore/revm-v55-upgrade`) first, and then merge this PR.

Reply:
And the pi hash from Euclid phase2 should be:
```
    /// Public input hash for a given chunk is defined as
    ///
    /// keccak(
    ///     chain id ||
    ///     prev state root ||
    ///     post state root ||
    ///     withdraw root ||
    ///     tx data hash ||
    ///     prev msg queue hash ||
    ///     post msg queue hash
    /// )
```

Reference: https://github.com/scroll-tech/zkvm-prover/blob/feat/phase2-misc-additions/crates/circuits/types/src/chunk/public_inputs.rs#L48-L58

Reply:
updated